lighthouse-core/lib/page-functions.js

```diff
@@ -438,40 +448,45 @@ function getBoundingClientRect(element) {
   };
 }
 
-/*
+/**
  * RequestIdleCallback shim that calculates the remaining deadline time in order to avoid a potential lighthouse
  * penalty for tests run with simulated throttling. Reduces the deadline time to (50 - safetyAllowance) / cpuSlowdownMultiplier to
  * ensure a long task is very unlikely if using the API correctly.
  * @param {number} cpuSlowdownMultiplier
- * @return {null}
+ * @return {void}
  */
 function wrapRequestIdleCallback(cpuSlowdownMultiplier) {
   const safetyAllowanceMs = 10;
   const maxExecutionTimeMs = Math.floor((50 - safetyAllowanceMs) / cpuSlowdownMultiplier);
   const nativeRequestIdleCallback = window.requestIdleCallback;
-  window.requestIdleCallback = (cb) => {
-    const cbWrap = (deadline, timeout) => {
+  window.requestIdleCallback = (cb, options) => {
+    /**
+     * @type {Parameters<typeof window['requestIdleCallback']>[0]}
+     */
+    const cbWrap = (deadline) => {
       const start = Date.now();
+      // @ts-expect-error - save original on non-standard property.
       deadline.__timeRemaining = deadline.timeRemaining;
       deadline.timeRemaining = () => {
-        return Math.min(
-          deadline.__timeRemaining(), Math.max(0, maxExecutionTimeMs - (Date.now() - start))
+        // @ts-expect-error - access non-standard property.
+        const timeRemaining = deadline.__timeRemaining();
+        return Math.min(timeRemaining, Math.max(0, maxExecutionTimeMs - (Date.now() - start))
         );
       };
       deadline.timeRemaining.toString = () => {
         return 'function timeRemaining() { [native code] }';
       };
-      cb(deadline, timeout);
+      cb(deadline);
     };
-    return nativeRequestIdleCallback(cbWrap);
+    return nativeRequestIdleCallback(cbWrap, options);
   };
   window.requestIdleCallback.toString = () => {
     return 'function requestIdleCallback() { [native code] }';
   };
 }
 
 /**
- * @param {HTMLElement} element
+ * @param {Element|ShadowRoot} element
```

Comment:
made the `HTMLElement` -> `Element` change for two reasons:

- if `element` is a `ShadowRoot`, `element.host` is an `Element` _not_ an `HTMLElement`, so would really have needed a second test to make sure it was an `HTMLElement`
- buuut, there doesn't appear to be any reason to limit `getNodeDetails` to `HTMLElement`...it seems reasonable for a gatherer to want to get `nodeDetails` of an SVG element, for instance

There aren't any significant effects from this change, just some type ones (like `__lighthouseNodesDontTouchOrAllVarianceGoesAway` becomes a `Map<Element, string>`)